implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java

```diff
@@ -125,6 +125,7 @@ public void start(ThreadFactory threadFactory) {
                 logger.error("unable to start stackdriver, service settings are not available");
             }
             else {
+                shutdownClientIfNecessary(true);
```

Comment:
I don't think this is necessary if nothing is done with the client in `stop`. The question is: should something be done in stop or only in close? We haven't been very clear about what should happen where (no JavaDocs). I'll open a separate issue to clear that up.

It's a bit wasteful of resources to not shutdown the client when a user calls `stop` on the registry, but in terms of publishing it shouldn't be an issue because the scheduled call to `publish` won't happen after `stop` is called until `start` is called again. Looking at other implementations, we do the same in WavefrontMeterRegistry - nothing special in `stop` and close the client on close after `super.close()`.

Reply:
> I don't think this is necessary if nothing is done with the client in `stop`.

I remembered now the issue I was pointing out in my previous review. We don't want to replace the client without shutting down the old one and making it eligible for garbage collection. Therefore having this line here should prevent that.

Reply:
As promised, I opened https://github.com/micrometer-metrics/micrometer/issues/4632 to track clarifying stop semantics.